lib/taskana-core-test/src/test/java/acceptance/jobs/HistoryCleanupJobAccTest.java

```diff
@@ -140,49 +132,66 @@ void should_CleanHistoryEventsUntilDate_When_SameParentBusinessTrueAndEventsQual
 
       TaskHistoryEvent eventToBeCleaned =
           createTaskHistoryEvent(
-              "wbKey1",
               "taskId1",
               TaskHistoryEventType.CREATED.getName(),
-              "wbKey2",
+              
               "someUserId",
               "someDetails",
               Instant.now().minus(20, ChronoUnit.DAYS),
               "sameParentId");
       historyService.create(eventToBeCleaned);
       TaskHistoryEvent eventToBeCleaned2 =
           createTaskHistoryEvent(
-              "wbKey1",
+              
               "taskId1",
               TaskHistoryEventType.COMPLETED.getName(),
-              "wbKey2",
+              
               "someUserId",
               "someDetails",
               Instant.now().minus(20, ChronoUnit.DAYS),
               "sameParentId");
       historyService.create(eventToBeCleaned2);
       TaskHistoryEvent eventToBeCleaned3 =
           createTaskHistoryEvent(
-              "wbKey1",
+              
               "taskId2",
               TaskHistoryEventType.CREATED.getName(),
-              "wbKey2",
+              
               "someUserId",
               "someDetails",
               Instant.now().minus(20, ChronoUnit.DAYS),
               "sameParentId");
       historyService.create(eventToBeCleaned3);
       TaskHistoryEvent eventToBeCleaned4 =
           createTaskHistoryEvent(
-              "wbKey1",
+              
               "taskId2",
               TaskHistoryEventType.CANCELLED.getName(),
-              "wbKey2",
+              
               "someUserId",
               "someDetails",
               Instant.now().minus(20, ChronoUnit.DAYS),
               "sameParentId");
       historyService.create(eventToBeCleaned4);
-
+      TaskHistoryEvent eventToBeCleaned5 =
+          createTaskHistoryEvent(
+              
+              "taskId3",
+              TaskHistoryEventType.CREATED.getName(),
+              "someUserId",
+              "someDetails",
+              Instant.now().minus(20, ChronoUnit.DAYS),
+              "sameParentId");
+      historyService.create(eventToBeCleaned5);
+      TaskHistoryEvent eventToBeCleaned6 =
```

Comment:
I think this testcase is correct. But I believe the more important test would be to test if a task history is cleaned if there is a DELETED event only and no COMPLETED, TERMINATED or CANCELLED event. Isn't this selection the bug/change?

Reply:
The COMPLETED and CANCELLED events are for different TaskIds. They should not affect the events with other TaskIds. There is no TERMINATED event in this test. There are no other Endstate events other than  DELETED for the Task with the id "taskId3". Would the test case with only the DELETED event still be necessary?

Reply:
Yes, I will add a test case with a unique parent id